### PR TITLE
[MM-19970] Add example usage of EnsureBot to the Demo Plugin

### PR DESCRIPTION
**Pull request by alxsah**
**Summary**
Add example usage of EnsureBot to the Demo Plugin. The existing implementation would be replaced to use EnsureBot with options. This is a follow up on https://github.com/mattermost/mattermost-server/pull/12153.

**Ticket Link**
https://mattermost.atlassian.net/browse/MM-19970
Fixes https://github.com/mattermost/mattermost-server/issues/13027

### CODE REVIEW

File: server/configuration.go [hunk 1]
```diff
@@ -203,36 +202,24 @@ func (p *Plugin) OnConfigurationChange() error {
...
+	}
+
+	iconImagePath := filepath.Join(path, "/assets/github.svg")
+
 	botId, ensureBotError := p.Helpers.EnsureBot(&model.Bot{
 		Username:    "demoplugin",
 		DisplayName: "Demo Plugin Bot",
 		Description: "A bot account created by the demo plugin.",
-	})
+	}, plugin.IconImagePath(iconImagePath))
```
Comment (hanzei) [inline server/configuration.go]: The argument of `plugin.IconImagePath` is a relative path inside the bundle, right? Hence you don't need https://github.com/mattermost/mattermost-plugin-demo/pull/76/files#diff-97fcb5163c5df39ace07e8d58616214eR205-R210.

Comment (alxsah) [reply server/configuration.go]: @hanzei Yes, sorry, you're right. I'll update this.

File: go.sum [hunk 2]
```diff
@@ -460,6 +521,8 @@ gopkg.in/yaml.v2 v2.0.0-20170812160011-eb3733d160e7/go.mod h1:JAlM8MvJe8wmxCU4Bl
 gopkg.in/yaml.v2 v2.2.1/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
 gopkg.in/yaml.v2 v2.2.2 h1:ZCJp+EgiOT7lHqUV2J862kp8Qj64Jo6az82+3Td9dZw=
 gopkg.in/yaml.v2 v2.2.2/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
+gopkg.in/yaml.v2 v2.2.3 h1:fvjTMHxHEw/mxHbtzPi3JCcKXQRAnQTBRo6YCJSVHKI=
+gopkg.in/yaml.v2 v2.2.3/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
```
Comment (hanzei) [inline go.sum]: Is it possible that you need to run `go mod tidy`?

**Comment by hanzei** [comment]
@prapti @ali-farooq0 I would like to merge this by the end of the week so we can use the new version of the demo plugin for release testing.

**Comment by ali-farooq0** [comment]
@prapti this doesn't introduce any _new_ changes, to confirm if this is working all you need to test is if the demo plugin is able to set the LHS icon, if it does, then this is 👌

**Comment by prapti** [comment]
@ali-farooq0 The log captures the following errors:
`RPC call MessageHasBeenPosted to plugin failed` when enabling the plugin and
`RPC call OnConfigurationChange to plugin failed.` when disabling the plugin. Would these be related to the changes introduced here? Haven't noticed such errors before.

**Comment by hanzei** [comment]
Interesting catch @prapti. They might be related. Would you please post the full logs here so we can debug this further?

**Comment by prapti** [comment]
@hanzei below is the error log when the demo plugin gets enabled. I copied the last 2 lines of the same error but every time it's enabled there's >50 lines of it:
`ERRO[2019-11-27T09:44:42.71106-08:00] RPC call MessageWillBePosted to plugin failed.  caller="mlog/log.go:174" error="connection is shut down" plugin_id=com.mattermost.demo-plugin
ERRO[2019-11-27T09:44:42.717113-08:00] RPC call MessageHasBeenPosted to plugin failed.  caller="mlog/log.go:174" error="connection is shut down" plugin_id=com.mattermost.demo-plugin
DEBU[2019-11-27T09:44:43.183342-08:00] Workers received config change.               caller="jobs/workers.go:108"`

The following is the error log when the demo plugin gets disabled:
`ERRO[2019-11-27T09:47:34.932787-08:00] RPC call OnConfigurationChange to plugin failed.  caller="mlog/log.go:174" error="connection is shut down" plugin_id=com.mattermost.demo-plugin
DEBU[2019-11-27T09:47:34.971162-08:00] Workers received config change.               caller="jobs/workers.go:108"`

Further observation: The errors don't show on the first installation and plugin enabling. But in order to see the error logs, you have to disable it, then every time you enable and disable there after, you'll see the logs. Please feel free to check the entire log out here: https://518.test.mattermost.cloud/admin_console/reporting/server_logs

**Comment by hanzei** [comment]
@prapti I can repo the issues you pointed out with both the master branch and this branch. Hence, I suppose it's unrelated. Would you be open on filling a JIRA ticket for this and continue with the review?

**Comment by hanzei** [comment]
Thank you for the PR @alxsah

**Comment by alxsah** [comment]
Thanks all!

**Pull request merged**